nature/static/nature/js/NatureOLMapWidget.js

```diff
@@ -0,0 +1,245 @@
+            var extent = ol.extent.createEmpty();
+            features.forEach(function(feature) {
+                this.featureOverlay.getSource().addFeature(feature);
+                ol.extent.extend(extent, feature.getGeometry().getExtent());
+            }, this);
+            // Center/zoom the map
+            this.map.getView().fit(extent, {maxZoom: this.options.default_zoom});
+        } else {
+            this.map.getView().setCenter(this.defaultCenter());
+        }
+        this.createInteractions();
+        if (initial_value && !this.options.is_collection) {
+            this.disableDrawing();
+        }
+        this.ready = true;
+    }
+
+    MapWidget.prototype.createMap = function() {
+        var projection = new ol.proj.Projection({
+            code: 'EPSG:3879',
+            extent: [21531406.93, 4503686.78, 25664437.76, 9371843.41],
+            units: 'm'
+        });
+        ol.proj.addProjection(projection);
+
+        var map = new ol.Map({
+            target: this.options.map_id,
+            layers: [this.options.base_layer],
+            view: new ol.View({
+                projection: projection,
+                zoom: this.options.default_zoom,
+                extent: [25440000.0, 6630000.0, 25571072.0, 6761072.0]
+            })
+        });
+        return map;
+    };
+
+    MapWidget.prototype.createInteractions = function() {
+        // Initialize the modify interaction
+        this.interactions.modify = new ol.interaction.Modify({
+            features: this.featureCollection,
+            deleteCondition: function(event) {
+                return ol.events.condition.shiftKeyOnly(event) &&
+                    ol.events.condition.singleClick(event);
+            }
+        });
+
+        // Initialize the draw interaction
+        var geomType = this.options.geom_name;
+        if (geomType === "Unknown" || geomType === "GeometryCollection") {
+            // Default to Point, but create icons to switch type
+            geomType = "Point";
+            this.currentGeometryType = new GeometryTypeControl({widget: this, type: "Point", active: true});
+            this.map.addControl(this.currentGeometryType);
+            this.map.addControl(new GeometryTypeControl({widget: this, type: "LineString", active: false}));
+            this.map.addControl(new GeometryTypeControl({widget: this, type: "Polygon", active: false}));
+            this.typeChoices = true;
+        }
+        this.interactions.draw = new ol.interaction.Draw({
+            features: this.featureCollection,
+            type: geomType
+        });
+
+        this.map.addInteraction(this.interactions.draw);
+        this.map.addInteraction(this.interactions.modify);
+    };
+
+    MapWidget.prototype.defaultCenter = function() {
+        var center = [this.options.default_lon, this.options.default_lat];
+        if (this.options.map_srid) {
+            return ol.proj.transform(center, 'EPSG:4326', this.map.getView().getProjection());
```

Comment:
Here `4326` is used instead of `3879`, is this working as intended?

Reply:
Good catch there. Actually we don't need the transformation here. I'll remove this part.

For some reason, Django uses `default_lon` and `default_lat` for all the map widgets, including OSMWidget which is using a different projection. I'll also rename the variable to be more precise.